pythran/analyses/range_values.py

```diff
@@ -0,0 +1,448 @@
+        Range(low=5, high=10)
+        >>> RangeValues.combine(Range(1, 5), Range(-5, -4), ast.Mult())
+        Range(low=-25, high=-4)
+        >>> RangeValues.combine(Range(-1, 5), Range(-5, 3), ast.Mult())
+        Range(low=-25, high=15)
+        >>> RangeValues.combine(Range(1, 5), Range(3, 8), ast.Mult())
+        Range(low=3, high=40)
+        >>> RangeValues.combine(Range(-1, 5), Range(3, 8), ast.Div())
+        Range(low=-1, high=1)
+        >>> RangeValues.combine(Range(-1, 5), Range(-5, -4), ast.Div())
+        Range(low=-2, high=0)
+        >>> RangeValues.combine(Range(-1, 5), Range(-5, 3), ast.Div())
+        Range(low=-inf, high=inf)
+        >>> RangeValues.combine(Range(-1, 5), Range(1, 13), ast.Mod())
+        Range(low=0, high=5)
+        >>> RangeValues.combine(Range(-21, 5), Range(1, 13), ast.Mod())
+        Range(low=0, high=13)
+        >>> RangeValues.combine(Range(1, 5), Range(-5, -4), ast.Pow())
+        Range(low=1.0, high=1.0)
+        >>> RangeValues.combine(Range(-1, 5), Range(-5, 3), ast.Pow())
+        Range(low=-1.0, high=125.0)
+        >>> RangeValues.combine(Range(1, 5), Range(3, 8), ast.Pow())
+        Range(low=1.0, high=390625.0)
+        >>> RangeValues.combine(Range(1, 5), Range(3, 8), ast.LShift())
+        Range(low=8, high=1280)
+        >>> RangeValues.combine(Range(1, float("inf")), Range(3, 8),
+        ...                       ast.LShift())
+        Range(low=8, high=inf)
+        >>> RangeValues.combine(Range(-float("inf"), 0), Range(3, 8),
+        ...                       ast.LShift())
+        Range(low=-inf, high=0)
+        >>> RangeValues.combine(Range(-3, 1), Range(3, float('inf')),
+        ...                       ast.LShift())
+        Range(low=-24, high=inf)
+        >>> RangeValues.combine(Range(10, 100), Range(3, 8), ast.RShift())
+        Range(low=0, high=12)
+        >>> RangeValues.combine(Range(10, float("inf")), Range(3, 8),
+        ...                       ast.RShift())
+        Range(low=0, high=inf)
+        >>> RangeValues.combine(Range(-float("inf"), 0), Range(3, 8),
+        ...                       ast.RShift())
+        Range(low=-inf, high=0)
+        >>> RangeValues.combine(Range(-30, 10), Range(3, float('inf')),
+        ...                       ast.RShift())
+        Range(low=-4, high=1)
+        >>> RangeValues.combine(Range(1, 5), Range(3, 8), ast.BitXor())
+        Range(low=-inf, high=inf)
+        >>> RangeValues.combine(Range(1, 5), Range(3, 8), ast.BitOr())
+        Range(low=-inf, high=inf)
+        >>> RangeValues.combine(Range(1, 5), Range(3, 8), ast.BitAnd())
+        Range(low=-inf, high=inf)
+        >>> RangeValues.combine(Range(-1, 5), Range(3, 8), ast.FloorDiv())
+        Range(low=-1, high=1)
+        >>> RangeValues.combine(Range(-1, 5), Range(-5, -4), ast.FloorDiv())
+        Range(low=-2, high=0)
+        >>> RangeValues.combine(Range(-1, 5), Range(-5, 3), ast.FloorDiv())
+        Range(low=-inf, high=inf)
+        """
```

Comment:
I will do it in the next commit to have it with "range combiners"... Should we have a special folder for this kind of code?